tests/5.2/target/test_target_map_device_pointer.c

```diff
@@ -0,0 +1,53 @@
+//===--- test_declare_target_map_device_pointer.c ------------------------------------------===//
+// 
+// OpenMP API Version 5.2 Nov 2021 
+//
+//This test checks if, after offloading, the initalized pointer variable retains its 
+//original value as per the semantics of the firstprivate clause. 
+//The pointer variable, nor the array it refers to, are not mapped to the device. This 
+//ensures that a matching mapped data item is not found for the pointer, so it retains
+//its original value.
+//===---------------------------------------------------------------------------------------===//
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <omp.h>
+#include "ompvv.h"
+
+#define N 1000
+
+int test_pointer() {
+  int compute_array[N];
+  int *p;
+  int errors = 0;
+  int i;
+  
+  // Array initialization
+  for (i = 0; i < N; i++)
+    compute_array[i] = i;
+  p = &compute_array[0];
+
+  #pragma omp target defaultmap(none:pointer)
```

Comment:
comment 1: remove defaultmap(none:pointer)
comment 2: how will this interact with implicit map of compute_array ? Maybe use a scalar?